pkg/informer/informer.go

```diff
@@ -374,21 +391,46 @@ func (d *DynamicDiscoverySharedInformerFactory) discoverTypes(ctx context.Contex
 		klog.V(4).Infof("Removing dynamic informer from maps for %q", gvr)
 		delete(d.informers, gvr)
 		delete(d.informerStops, gvr)
+		delete(d.startedInformers, gvr)
 	}
 
-	d.gvrs = latest
-
 	return nil
 }
 
+func (d *DynamicDiscoverySharedInformerFactory) Start(stop <-chan struct{}) {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+
+	for gvr, informer := range d.informers {
+		if !d.startedInformers[gvr] {
+			go informer.Informer().Run(stop)
```

Comment:
am not happy with the lifecycle semantic of this informer. When stop closed, and this could be long before the whole ddsif context is done, we leak some stopped informer here. Either we have to remove it from the ddsif, or we should just ignore the stop channel and create a fresh one and place it in the stop channel map.